pkg/secrethub/client.go

```diff
@@ -1,19 +1,17 @@
 package secrethub
 
 import (
-	awssdk "github.com/aws/aws-sdk-go/aws"
 	"github.com/secrethub/secrethub-go/internals/api"
-	"github.com/secrethub/secrethub-go/internals/auth"
-	"github.com/secrethub/secrethub-go/internals/aws"
 	"github.com/secrethub/secrethub-go/internals/crypto"
 	"github.com/secrethub/secrethub-go/internals/errio"
+	"github.com/secrethub/secrethub-go/pkg/secrethub/credentials"
+	"github.com/secrethub/secrethub-go/pkg/secrethub/http"
 )
 
 // Client is the SecretHub client.
-type Client interface {
+type ClientAdapter interface {
```

Comment:
As packages should consume the interface and not the struct, I'd prefer to use the nice name `secrethub.Client` for the interface.

Also, the current changeset can have some none intuitive error messages, as `secrethub.Client` now points to the struct (without pointer) instead of the interface.

Reply:
Can you please clarify your second point? (example can suffice)

Reply:
https://github.com/secrethub/secrethub-cli/blob/v0.21.0/internals/secrethub/app.go#L81 used to accept a function that returns the interface, but now accepts a function that returns the struct (which does not implement the interface).

Reply:
Okay. That's clear. Not sure how big of a problem it is though. Do you have a suggestion of what else to call it?